build/sharpmake/src/main.cs

```diff
@@ -291,6 +293,65 @@ private static void GenerateCompilerDatabases()
     }
   }
 
+  private static void AutoGenerateEnums()
+  {
+    foreach (EnumGenerationSettings enum_gen_settings in GenerateSettings.EnumsToAutoGenerate.Values)
+    {
+      WriteAutoGeneratedEnumToFile(enum_gen_settings.ClassName, enum_gen_settings.ProjectToEnumValues, enum_gen_settings.Filepath);
+    }
+  }
+
+  private static void WriteAutoGeneratedEnumToFile(string className, Dictionary<string, List<string>> enumValues, string filename)
+  {
+    StringBuilder sb = new StringBuilder();
+    WriteCustomGenerationHeader(sb);
+
+    sb.AppendLine($"  enum class {className}");
+    sb.AppendLine("  {");
+
+    foreach (var project_tags in enumValues)
+    {
+      sb.AppendLine($"    // {className} values for {project_tags.Key}");
+
+      foreach (string tag in project_tags.Value)
+      {
+        sb.AppendLine($"    {tag},");
+      }
+    }
+
+    sb.AppendLine("  };");
```

Comment:
Wouldn't it be better to use an ISourceGenerator here?
This would make the code a bit more readable I think.

https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.isourcegenerator?view=roslyn-dotnet-4.3.0
https://learn.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview

Reply:
from a quick read source generator is to generate C# code not C++ code

Reply:
You are correct, my bad.
But I still think this part of the code can become a bit more readable we should not apply it within this PR but maybe come up with something to generate C++ code. 

This is probably not the last code that we will generate, will be useful to have a system for that.